packages/httpServer/dependencies/swagger/createSwaggerDefinition/index.js

```diff
@@ -37,26 +39,33 @@ export default () => (routes, { title, description, contact, version }) => {
   }
 }
 
-const simplePathParam = name => ({
-  name,
-  in: 'path',
-  description: `${name} path param`,
-  required: true,
-  style: 'simple',
-  schema: {
-    type: 'string'
+const definePathParam = ({ name, type, flags }) => {
+  const { description, ...metadata } = flags
+  const detailedDescription = description ?
+    `${description}  ---  Additional Parameter Context: ${JSON.stringify(metadata)}` :
+    `${name} path param`
+  return {
+    name,
+    in: 'path',
+    description: detailedDescription,
+    required: true,
+    style: 'simple',
+    schema: { type }
   }
-})
+}
 
-const getPathParams = path => {
-  const params = []
+const getPathParams = ({ path, schema = {}}) => {
   const parts = path.split(PATH_DELIMITER)
-  parts.forEach(part => {
-    if (part[0] === PARAM_DELIMITER) {
-      params.push(part.substring(1))
-    }
-  })
-  return params.map(simplePathParam)
+  const pathSchema = has(schema, `${PATH_PARAMS}`) ?
```

Comment:
Do you need to force this to a string? Is that why you're doing &#96;${PATH_PARAMS}&#96;